src/guardianSigner.ts

```diff
@@ -0,0 +1,47 @@
+import { ISignable, ISignature } from "./interface";
+import { UserSecretKey } from "./userKeys";
+import { UserSigner } from "./userSigner";
+import { Signature } from "./signature";
+import { ErrSignerCannotSign } from "./errors";
+
+/**
+ * ed25519 signer
+ */
+export class GuardianSigner extends UserSigner {
+    private readonly guardianSecretKey: UserSecretKey;
```

Comment:
you can reuse UserSigner`s member `secretKey` (you will have to make it protected) and not redefine one extra member for GuardianSigner.

Reply:
true,
I wanted to delimit as much as possible the idea of userSecretKey from the guardianSecreyKey but yes, it's smoother to remove it.